src/main/java/com/synopsys/integration/alert/component/settings/descriptor/SettingsUIConfig.java

```diff
@@ -197,4 +211,31 @@ public SettingsUIConfig() {
         return List.of();
     }
 
+    private Collection<String> validateMetaDataUrl(FieldValueModel fieldToValidate, FieldModel fieldModel) {
+        Optional<FieldValueModel> samlEnabledField = fieldModel.getFieldValueModel(SettingsDescriptor.KEY_SAML_ENABLED);
+        boolean samlEnabled = samlEnabledField.flatMap(field -> field.getValue())
+                                  .map(value -> Boolean.valueOf(value))
+                                  .orElse(false);
+        if (samlEnabled) {
+            if (!fieldToValidate.hasValues() && !filePersistenceUtil.exists(SettingsDescriptor.SAML_METADATA_FILE)) {
+                return List.of(SettingsDescriptor.FIELD_ERROR_SAML_METADATA_URL_MISSING);
+            }
+        }
+        return List.of();
+    }
+
+    private Collection<String> validateMetaDataFile(FieldValueModel fieldToValidate, FieldModel fieldModel) {
+        Optional<FieldValueModel> samlEnabledField = fieldModel.getFieldValueModel(SettingsDescriptor.KEY_SAML_ENABLED);
+        boolean samlEnabled = samlEnabledField.flatMap(field -> field.getValue())
+                                  .map(value -> Boolean.valueOf(value))
+                                  .orElse(false);
+        if (samlEnabled) {
```

Comment:
What will happen if someone uploads a file without enabling SAML? Will a file upload but we simply do nothing with it?